tensorflow/compiler/tf2tensorrt/trt_convert.h

```diff
@@ -0,0 +1,119 @@
+  // expected for every tensor in the graph (excluding those which will be
+  // fused). If a range is missing, an error will occur. Please note that
+  // accuracy may be negatively affected if there is a mismatch between which
+  // tensors TRT quantizes and which tensors were trained with fake
+  // quantization.
+  bool use_calibration = true;
+
+  // Whether to enable dynamic shape mode for the TRT engines. Enabling
+  // dynamic shape mode can also improve the conversion rate of graphs with
+  // static input shape.
+  bool use_dynamic_shape = true;
+
+  // In dynamic shape mode we create an engine that can handle various input
+  // shape ranges. We use the input provided by the user during build time
+  // to derive the shape optimization profiles for the TRT engines in the graph.
+  // One can have different strategies for deriving these profiles:
+  // - `kRange`: create one profile that works for inputs with dimension values
+  //   in the range of [min_dims, max_dims] where min_dims and max_dims are
+  //   derived from the provided inputs.
+  // - `kOptimal`: create one profile for each input. The profile only works for
+  //   inputs with the same dimensions as the input it is created for. The GPU
+  //   engine will be run with optimal performance with such inputs.
+  // - `kRangeOptimal`: create the profiles for both `Range` and `Optimal`.
+  // - `kImplicitBatchModeCompatible`: create the profiles that will produce the
+  //   same GPU engines as the implicit_batch_mode would produce.
+  ProfileStrategy profile_strategy = ProfileStrategy::kRange;
+
+  // Whether to allow bulding TRT engines at runtime. If no TensorRT engine can
+  // be found in cache that can handle the given inputs during runtime, then a
+  // new TensorRT engine is built at runtime if allow_build_at_runtime=True,
+  // otherwise native TF is used. We recommend to set it false and build the
+  // engine in advance, to avoid runtime overhead.
+  bool allow_build_at_runtime = true;
+
+  // Record the TRT engine as an attribute for the TRTEngineOp. This is only
+  // valid when max_cached_engines == 1. Note: the frozen graph together with
+  // the serialized engines have to be below 2GiB (protobuf size limit). If
+  // convert_to_static_engine = false, then the converted graph_def only
+  // contains placeholder TRTEngineOp nodes, and the TRT engines will be created
+  // when we run the
+  bool convert_to_static_engine = true;
+};
+
+/**
+ * Converts the graph with TF-TRT.
+ *
+ * If inputs is not empty, then engines will be built using the provided inputs.
+ *
+ * Arguments:
+ * - frozen_graph_def input graph, it is assumed to be frozen.
+ * - signature
+ * - conv_params parameters for the TF-TRT conversion
+ * - inputs Graph inputs that we will use for building the TRT engines.
+ *
+ * Returns the converted graph_def.
+ */
+StatusOr<GraphDef> ConvertAndBuild(
+    const GraphDef& frozen_graph_def, const SignatureDef& signature,
```

Comment:
We may want to replace SignatureDef with 
std::vector<string>& input_names, const std::vector<string>& output_names